### PR TITLE
Bring up to date, and divest from scala.util.parsing

### PR DESCRIPTION
**Pull request by mossprescott**
@sellout: consider yourself assigned.

### CODE REVIEW

File: src/main/scala/invertible/Position.scala [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  private def column(offset: Int): Int =
+    offset - lineIndex(line(offset) - 1) + 1
+
+  private def lineContents(line: Int): String =
+    chars.subSequence(lineIndex(line-1), lineIndex(line)-1).toString
+}
+
+object Position {
+  implicit val semigroup: Semigroup[Position] = new Semigroup[Position] {
+    def append(p1: Position, p2: => Position) = Position(p1.chars, p1.startOffset, p2.endOffset)
```
Comment (sellout) [inline src/main/scala/invertible/Position.scala:90]: Does this actually satisfy the semigroup laws?

Comment (sellout) [inline src/main/scala/invertible/Position.scala:90]: We can’t actually test the laws until scalaz-specs2 works with scalaz 7.2 … unless you want to copy over the `CheckAll` class, like Matryoshka did.

**Comment by sellout** [comment]
Looks good other than the questionable semigroup.

File: src/main/scala/invertible/Position.scala (hunk 1 above)
Comment (mossprescott) [inline src/main/scala/invertible/Position.scala:90]: It does seem dodgy, doesn't it? Actually, it is associative as written, although there's no good reason _not_ to fix it to also commute. 

The real problem with it is that there's no requirement that two Positions actually refer to the same source. I suppose the source could be pulled out of `Position` altogether and then we would really just be tracking something like a pair of Ints for each AST node. But then we would have to track the source separately, which would make the annotations a lot less useful. Any ideas?

Comment (sellout) [inline src/main/scala/invertible/Position.scala:90]: It was the possible source discrepancy that bothered me, but yeah, seems lawful as is. I don’t think it’s ideal, but I don’t have any better suggestions.

**Comment by sellout** [comment]
Alright, 👍🏾.

**Pull request merged**